spec/mongoid/association/referenced/has_and_belongs_to_many/proxy_spec.rb

```diff
@@ -3770,4 +3770,11 @@ class Distributor
       expect(p2.d_ids).to match_array([d2.id])
     end
   end
+
+  context "when there is a foreign key in the aliased associations" do
+    it "has the correct aliases" do
+      expect(Dog.aliased_associations["breed_ids"]).to eq("breeds")
+      expect(Breed.aliased_associations["dog_ids"]).to eq("dogs")
```

Comment:
> In `Address.aliased_associations has user_ids => user` why is user singular?

@p-mongo if you look here, breeds and dogs are not singular? Where are you seeing user being singular?